platypus/core.py

```diff
@@ -987,9 +987,11 @@ def add(self, solution):
         else:
             self._contents = list(itertools.compress(self._contents, nondominated)) + [solution]
 
-            if dominated and not_same_box:
+            if any(dominated) and all(not_same_box):
```

Comment:
Might also consider getting rid of the dominated check.

In the MOEA Framework, which is the precursor to this library, I actual split out two types of improvement:

https://github.com/MOEAFramework/MOEAFramework/blob/28d11977f5cf0a52881b6862f265009bb0dd9f55/src/org/moeaframework/core/EpsilonBoxDominanceArchive.java#L46-L55

Reply:
Yeah, I think to keep the two libraries consistent, non-dominated solutions should count as improvements.